### PR TITLE
[Improve][Connector-V2][Fake] Refactor fake connector e2e test cases

### LINKED ISSUE #3602: [Improve][Connector-V2][Fake] Refactor Fake connector e2e test cases
sub-task of #3601

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #3602 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/resources/fake_to_assert.conf [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  }
+}
+
+sink {
+  Assert {
+    rules {
+      row_rules = [
+        {
+          rule_type = MAX_ROW
+          rule_value = 5
```
Comment (TaoZex) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/resources/fake_to_assert.conf:74]: Do we need to add MIN_ROW？

Comment (TyrantLucifer) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-fake-e2e/src/test/resources/fake_to_assert.conf:74]: I think it's meaningless

**Review by TaoZex — DISMISSED** [review]

**Review by EricJoy2048 — DISMISSED** [review]
LGTM, Please waiting CI complete.

**Review by TaoZex — APPROVED** [review]
+1

**Review by hailin0 — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]

**Pull request merged**